src/pip/_internal/locations/base.py

```diff
@@ -42,9 +42,11 @@ def get_src_prefix() -> str:
 try:
     # Use getusersitepackages if this is present, as it ensures that the
     # value is initialised properly.
-    user_site: typing.Optional[str] = site.getusersitepackages()
+    user_site = site.getusersitepackages()
 except AttributeError:
-    user_site = site.USER_SITE
+    _user_site = site.USER_SITE
+    assert _user_site is not None
+    user_site = _user_site
```

Comment:
Same here, this value can be None if the user does not have the user site enabled.